### DISCUSSION TITLE
1.1.7

### DISCUSSION THREAD

**Discussion by benhillis** [microsoft/WSL#9898] [General]
*  Fix race condition in wslhost's com server logic to prevent a crash
*  Remove cgroup_no_v1=all from kernel command line (solves #9868)
*  Remove unused WSL_E_PACKAGE_UPGRADE_AVAILABLE error message (solves #9866)
*  Give an actionable error message if an error occurs in reading reg keys (solves #9834)
*  Redesign the way that page reporting is configured, will result on lower vmmemWSL usage on newer versions of Windows
    - Removes wsl2.pageReportingOrder option from `%userprofile%\.wslconfig`
*  Re-register package if com class creation fails when talking to wslservice (solves #9064)
*  Remove bond, dummy, fb_tunnel adapters via kernel command line (solves #7791)
*  Bind mount /run/user to /<drvfs-prefix>/wsl/user before the namespaces fork
*  Make /run/user/<uid> an overlayfs so each user can write its folder
*  Wait for systemd to be started before forking the mount namespaces
*  Update Microsoft.WSL.Kernel to [6.1.21.1-1](https://github.com/microsoft/WSL2-Linux-Kernel/releases/tag/linux-msft-wsl-6.1.21.1)
    - Update to stable kernel version v6.1.21
    - Out-of-tree vpci patches now included from upstream
    - Out-of-tree vsock patches now included from upstream
    - Enable anonymous VMA name support (ANON_VMA_NAME)
    - Enable use nosuid,noexec mount options on devtmpfs (DEVTMPFS_SAFE)
    - Enable register zeroing on function exit (ZERO_CALL_USED_REGS)
    - Enable check for invalid mappings in user page tables (PAGE_TABLE_CHECK)
    - Enable Kernel Function Probe (FPROBE)
*  Update Microsoft.WSLg to version [1.0.51](https://github.com/microsoft/wslg/releases/tag/v1.0.51)
    - rdpapplist/rdp plugin: support pin WSLg app icon to taskbar (require msrdc update)
    - rdp backend: handle the case nothing to send for window zorder to client
    - rdp backend: use rdp_dispatch_task_to_display_loop for dispatching failure response to display loop
    - rdp shell: fix overactive assert at launch_desktop_shell_process
    - rdp shell/rdp backend: support pin WSLg app icon to taskbar (requires msrdc update)

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WSL/releases/tag/1.1.7'>1.1.7</a>.</em>

**Comment by JYInMyHeart** [comment]
I've seen some features require msrdc update. plz let me know how can I update msrdc.

**Comment by Morrigan-Ship** [comment]
what is `wsl2.pageReporting` alternative ?
